metricflow/specs/python_object.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import List
+
+from dbt_semantic_interfaces.call_parameter_sets import ParseWhereFilterException
+from dbt_semantic_interfaces.implementations.filters.where_filter import PydanticWhereFilter
+
+from metricflow.naming.linkable_spec_name import StructuredLinkableSpecName
+from metricflow.protocols.query_parameter import GroupByParameter
+from metricflow.query.query_exceptions import InvalidQueryException
+from metricflow.specs.query_param_implementations import DimensionOrEntityParameter, TimeDimensionParameter
+
+
+def parse_object_builder_naming_scheme(group_by_item_name: str) -> GroupByParameter:
```

Comment:
I don't know how much of this still applies and what's been improved. Agreed we should check in with @DevonFulcher, probably worth doing that before merge since you'll need to update some stuff due to me mucking about with WhereFilter interfaces in dbt-semantic-interfaces.

Reply:
Hey, sorry I am just catching up here. In general, I agree that query interface objects should share more code. Now that they all share an interface, this should be easier. I'm happy to chat about that more about that.

Reply:
Chatted with @DevonFulcher - using this for now until an improved setup is ready.